gwpy/cli/cliproduct.py

```diff
@@ -659,13 +658,22 @@ def annotate_save_plot(self, arg_list):
 
         # image plots don't have legends
         if not self.is_image():
-            leg = self.ax.legend(prop={'size': 10})
+            leg = self.plot.gca().legend(prop={'size': 8})
             # if only one series is plotted hide legend
             if self.n_datasets == 1 and leg:
                 try:
                     leg.remove()
                 except NotImplementedError:
                     leg.set_visible(False)
+            elif self.n_datasets <= 5 and leg:
+                self.plot.gca().legend(loc='best')
+            elif self.n_datasets > 5 and leg:
+                self.plot.subplots_adjust(right=0.75)
+                #self.plot.gca().legend(bbox_to_anchor=(1.0, 1.0))
+                self.plot.gca().legend(bbox_to_anchor=(1.5, 1), loc='right', borderaxespad=0.)
```

Comment:
@areeda, I think it makes more sense here to use 

```python
self.plot.gca().legend(bbox_to_anchor=(1.001, 1), loc='left', borderaxespad=0)
```

to left-align the legend to the right-hand axes, rather than right-aligning to the edge of the figure (which I think is what you are doing now).